funcx_web_service/__init__.py

```diff
@@ -62,6 +65,57 @@ def create_tables():
         db.init_app(application)
         db.create_all()
 
+    # this is called before every request
+    @application.before_request
+    def before_request():
+        # this is basic request data
+        extra = {
+            "request_json": request.json,
+            "path": request.path,
+            "full_path": request.full_path,
+            "method": request.method,
+            "type": "before_request"
+        }
+        # this is additional data passed into the request URL via
+        # the view arguments (e.g. in '/tasks/<task_id>', the value of
+        # task_id is a view argument, so this will make task_id a
+        # key in the logged JSON object)
+        if request.view_args:
+            extra.update(request.view_args)
+        logger.info("before_request", extra=extra)
+
+    # this is called only after requests that do not raise an exception
+    @application.after_request
+    def after_request(response):
+        # this is basic request and response data
+        extra = {
+            "request_json": request.json,
+            "response_json": response.json,
```

Comment:
While logging the complete request and response is very useful for debugging, we should be mindful of a couple of issues:

- Payload size.  Cloudwatch has a maximum event size of 256 KB.  This means if the response (e.g. getting the result from function) or the request (e.g. posting a large batch) is larger than that, this log will either be dropped or split across multiple eventns making it hard to track.  
- Malformed JSON.  Request.json will raise "BadRequest" exception here, and that will both drop the log and abort the request with a 500 error.
- Security and compliance.  This is probably a lower risk issue at this point in FuncX's life, but eventually I'd expect these responses may contain results with sensitive information we don't want stored in logs.  

To mitigate these issues, I'd do a few things:

1. Check the length of the request and response body before including them here.
2. Add a handler for json decoding exceptions that either leaves `request_json` or `response_json` unset, or set's them to None.

Reply:
@joshbryan-globus I made some basic changes to account for this:
- try block around grabbing the json to avoid exceptions
- basic check that content_size of request/response does not exceed 100,000 Bytes before logging these

Does that seem good for now?

Reply:
Yeah, that looks safer.  Thanks!